### PR TITLE
update eager code gen

### PR DESCRIPTION
**Pull request by phlrain**
### PR types
Breaking changes

### PR changes
OPs

### Describe
update

### CODE REVIEW

File: paddle/fluid/eager/utils.cc [hunk 1]
```diff
@@ -327,6 +335,18 @@ paddle::experimental::Tensor EagerUtils::RecoverTensorWrapper(
   return tw->recover(grad_node);
 }
 
+paddle::optional<const paddle::experimental::Tensor&>
+EagerUtils::RecoverOptionalTensorWrapper(
+    TensorWrapper* tw, const std::shared_ptr<GradNodeBase>& grad_node) {
+  auto tmp = tw->recover(grad_node);
```
Comment (JiabinYang) [inline paddle/fluid/eager/utils.cc:345]: PADDLE_ENFORCE_NOT_NULL for tw

File: paddle/fluid/eager/utils.cc [hunk 2]
```diff
@@ -327,6 +335,18 @@ paddle::experimental::Tensor EagerUtils::RecoverTensorWrapper(
...
+
+  paddle::optional<const paddle::experimental::Tensor&> res = paddle::none;
```
Comment (JiabinYang) [inline paddle/fluid/eager/utils.cc]: How about construct as none

**Review by JiabinYang — DISMISSED** [review]
polish code later

**Review by chenwhql — DISMISSED** [review]
LGTM for PADDLE_ENFORCE

**Review by JiabinYang — DISMISSED** [review]
LGTM

**Review by JiabinYang — APPROVED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]

**Pull request merged**